src/Table/TableController.cc

```diff
@@ -234,6 +251,28 @@ void TableController::OnFileListRequest(
                 stat(entry.path().c_str(), &file_stats);
                 file_info->set_date(file_stats.st_mtim.tv_sec);
             }
+
+            ++num_of_files_done;
+            percentage = (float)num_of_files_done / (float)total_files;
+            auto current_time = std::chrono::high_resolution_clock::now();
+            auto dt = std::chrono::duration<double>(current_time - start_time).count();
+
+            auto report_progress = [&]() {
```

Comment:
I don't like the idea of redefining a lambda inside the loop. Why does it need to be in the loop, instead of outside?

Reply:
`current_time` could easily be a parameter